runtime/engine/src/main/java/io/aklivity/zilla/runtime/engine/Engine.java

```diff
@@ -387,5 +391,19 @@ public void onError(
         {
             errorHandler.onError(error);
         }
+
+        @Override
+        public LongSupplier counter(
+            String namespace,
+            String binding,
+            String metric)
+        {
+            if (metrics == null)
+            {
+                return () -> 0L;
+            }
+            MetricRecord record = metrics.findRecord(namespace, binding, metric);
+            return () -> record == null ? 0L : record.value();
+        }
```

Comment:
We need to do more of a review on `MetricRecord` etc before we consider pulling it up into `engine`. Please move it back into `command-metrics` for now.

In the meantime, let's simplify.
 - lookup `bindingId` and `metricId` based on provided `namespace`, `binding` and `metric` names from label manager
 - keep the dispatchers parameter to the constructor and retain in a field
 - walk the dispatchers to find a `LongSupplier` for each `counter`
 - in DispatcherAgent.supplyCounter(bindingId, metricId), return countersLayout.suppyReader(bindingId, metricId)
 - return an aggregating `LongSupplier` lambda that sums all the `LongSupplier`s to get the total counter value